radicle-cli/src/commands/issue.rs

```diff
@@ -0,0 +1,247 @@
+                    state = Some(State::Open);
+                }
+                Long("solved") if op == Some(OperationName::State) => {
+                    state = Some(State::Closed {
+                        reason: CloseReason::Solved,
+                    });
+                }
+                Long("reaction") if op == Some(OperationName::React) => {
+                    if let Some(emoji) = parser.value()?.to_str() {
+                        reaction =
+                            Some(Reaction::from_str(emoji).map_err(|_| anyhow!("invalid emoji"))?);
+                    }
+                }
+                Long("description") if op == Some(OperationName::Create) => {
+                    description = Some(parser.value()?.to_string_lossy().into());
+                }
+                Value(val) if op.is_none() => match val.to_string_lossy().as_ref() {
+                    "n" | "new" => op = Some(OperationName::Create),
+                    "s" | "state" => op = Some(OperationName::State),
+                    "d" | "delete" => op = Some(OperationName::Delete),
+                    "l" | "list" => op = Some(OperationName::List),
+                    "r" | "react" => op = Some(OperationName::React),
+
+                    unknown => anyhow::bail!("unknown operation '{}'", unknown),
+                },
+                Value(val) if op.is_some() => {
+                    let val = val
+                        .to_str()
+                        .ok_or_else(|| anyhow!("issue id specified is not UTF-8"))?;
+
+                    id = Some(
+                        IssueId::from_str(val)
+                            .map_err(|_| anyhow!("invalid issue id '{}'", val))?,
+                    );
+                }
+                _ => {
+                    return Err(anyhow!(arg.unexpected()));
+                }
+            }
+        }
+
+        let op = match op.unwrap_or_default() {
+            OperationName::Create => Operation::Create { title, description },
+            OperationName::State => Operation::State {
+                id: id.ok_or_else(|| anyhow!("an issue id must be provided"))?,
+                state: state.ok_or_else(|| anyhow!("a state operation must be provided"))?,
+            },
+            OperationName::React => Operation::React {
+                id: id.ok_or_else(|| anyhow!("an issue id must be provided"))?,
+                reaction: reaction.ok_or_else(|| anyhow!("a reaction emoji must be provided"))?,
+            },
+            OperationName::Delete => Operation::Delete {
+                id: id.ok_or_else(|| anyhow!("an issue id to remove must be provided"))?,
+            },
+            OperationName::List => Operation::List,
+        };
+
+        Ok((Options { op }, vec![]))
+    }
+}
+
+pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+    let profile = ctx.profile()?;
+    let signer = term::signer(&profile)?;
```

Comment:
we first load `id` and then re-load `repo`... `cwd()` returns repo but it's `git2::Repository`.

if this is annoying, I change `cwd()` to return our `Repository` and then we can get `.raw()`